### PR TITLE
Add test requirements including minimal versions

### PR DESCRIPTION
**Pull request by sebix**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -42,4 +42,8 @@
         'Programming Language :: Python :: 3 :: Only',
         'Programming Language :: Python :: Implementation :: PyPy',
     ],
+    tests_require=[
```
Comment (ekohl) [inline setup.py:45]: This is deprecated since setuptools 41.5.0 per https://github.com/pypa/setuptools/pull/1878

**Comment by ekohl** [comment]
Any thoughts on this? I also think that 6.0.0 is not entirely correct. From the changelog:
> LogCapture.actual() is now part of the documented public interface.

However, the class is much older. 1.2.0 mentions that `LogCapture` can now do some things, which means the class has always been there.

I think this can be closed.

**Review by ekohl — CHANGES_REQUESTED** [review]
Marking as changes required to indicate it's been reviewed.

**Comment by florisla** [comment]
The preferred way to do this in `setup.py` is to use `extras_require`.

    extras_require={
        'test': ['pytest>=3.4.0']
    }

**Comment by sebix** [comment]
As requested, I changed the testfixtures version to 1.2.0 and `tests_require` to `extras_require`

**Comment by florisla** [comment]
Hi,

The required version of `testfixtures` in `tox.ini` differs versus `setup.py`. Any particular reason?

Could you squash your three commits into one? We don't need the `test_require` commits.

**Comment by florisla** [comment]
I'm OK with putting this 'feature' in.

It's not the official testing strategy (that's Tox, and CI) but this is more light-weight and can be handy for contributors.

**Comment by sebix** [comment]
I think I incorporated all requested changes now

**Comment by florisla** [comment]
Thanks!

File: setup.py [hunk 2]
```diff
@@ -42,4 +42,9 @@
         'Programming Language :: Python :: 3 :: Only',
         'Programming Language :: Python :: Implementation :: PyPy',
     ],
+    extras_require={
+        'test': ['testfixtures>=1.2.0',
+                 'pytest>=3.4.0',
+                 ],
```
Comment (ekohl) [inline setup.py:48]: Nit: I'd prefer to format this similar to other lines
```suggestion
        'test': [
          'testfixtures>=1.2.0',
          'pytest>=3.4.0',
        ],
```

**Pull request merged**